sam/service_accounts.tf

```diff
@@ -0,0 +1,48 @@
+# Main service account for the sam service.
+# Note that Sam's service account is on the classic storage google project. This is to make it easier for Sam to use the
+# right project for storage with all of its current uses of service accounts. That may not be a compelling reason in the
+# future.
+resource "google_service_account" "sam" {
```

Comment:
Is adding an option to either use an existing service account or create a new one something that we might want to add?

For an example see:
https://github.com/broadinstitute/terraform-ap-modules/blob/b8fe28d4cbcfeb199b4a1a402f4d1a34109b4a29/single-cell-portal/sa.tf#L1

Similarly, do we want to add toggles for creating all the persistence that we are currently re-using? All of this can probably be done in a later revision of this module, but I wanted to be sure we're on the same page regarding the path forward. I see this module as eventually supporting both the use case of creating the infrastructure for a k8s SAM pointing to the state of an app-server SAM, as well as the use case of standing up a standalone k8s SAM with all the required state/infrastructure.

Reply:
Yeah, thinking about it now I think we may want to re-use service accounts between different developers personal environments most of the time. OTOH, it's a little more complicated than always creating service accounts. I guess how much does it matter if we have many service accounts?

I think we will definitely want to add toggles or more modules for creating persistence. I'm open to having it all in this module or in separate modules.